File: apps/legacy_vault/legacy_vault/Core/Models/ValueKeyword.swift
import Foundation

struct ValueKeyword: Identifiable, Codable, Hashable {
    let id: UUID
    var word: String
    var frequency: Int
    var firstOccurrence: Date
    var lastOccurrence: Date
    var category: KeywordCategory

    enum KeywordCategory: String, Codable {
        case family, career, emotion, challenge, peace, growth
    }

        func hash(into hasher: inout Hasher) {
        hasher.combine(id)
    }

    static func == (lhs: ValueKeyword, rhs: ValueKeyword) -> Bool {
        lhs.id == rhs.id
    }
}

File: archive/legacy_quarantine/apps/legacy_vault_android/app/src/main/java/com/sfx/legacyvault/core/models/ValueKeyword.kt
package com.sfx.legacyvault.core.models

import java.util.UUID

data class ValueKeyword(
    val id: String = UUID.randomUUID().toString(),
    val word: String = "",
    val frequency: Int = 0,
    val firstOccurrence: Long = 0L,
    val lastOccurrence: Long = 0L,
    val category: String = ""
)
